gms_process_inputs/somatic_exome.process_inputs.pl

```diff
@@ -0,0 +1,184 @@
+    or die 'no reference found for input';
+$inputs->{reference} = $ref_build->full_consensus_path('fa');
+my (@normal_bams, @normal_rg_fields, @tumor_bams, @tumor_rg_fields);
+
+my ($tumor_input, @extra) = grep { $_->name eq 'tumor_sample' } @inputs;
+if (@extra) {
+    die 'multiple inputs found for tumor sample';
+}
+
+
+my $tumor_sample = $tumor_input->value_class_name->get($tumor_input->value_id)
+    or die 'no tumor found for input';
+
+my ($normal_input, @extra) = grep { $_->name eq 'normal_sample' } @inputs;
+if (@extra) {
+    die 'multiple inputs found for normal sample';
+}
+
+my $normal_sample = $normal_input->value_class_name->get($normal_input->value_id)
+    or die 'no normal found for input';
+
+my @instrument_data_inputs = grep { $_->name eq 'instrument_data' } @inputs;
+my @ids = map $_->value_id, @instrument_data_inputs;
+
+my $target_region_set_name;
+my $multiple_trsn = 0;
+
+for my $input (@instrument_data_inputs) {
+    my $id = $input->value_class_name->get($input->value_id)
+        or die 'no instrument data found for input';
+    
+    my $pu = join('.', $id->flow_cell_id, $id->lane, ( $id->can('index_sequence')? $id->index_sequence : () ));
+    my $sm = $id->sample->name;
+    my $lb = $id->library->name;
+    my $pl = 'Illumina';
+    my $cn = 'WUGSC';
+    my $rgid = $id->id;
+
+    my $bam_path = $id->bam_path
+      or die 'no bam path found for instrument data ' . $id->id;
+    
+    if($id->sample eq $tumor_sample){
+      push @tumor_bams, {class => 'File', path => $bam_path};
+      push @tumor_rg_fields, join("\t", '@RG', "ID:$rgid", "PU:$pu", "SM:$sm", "LB:$lb", "PL:$pl", "CN:$cn");
+    } elsif($id->sample eq $normal_sample) { 
+      push @normal_bams, {class => 'File', path => $bam_path};
+      push @normal_rg_fields, join("\t", '@RG', "ID:$rgid", "PU:$pu", "SM:$sm", "LB:$lb", "PL:$pl", "CN:$cn");
+    } else {
+       die "no sample match found:" . $id->id;
+    }
+
+    unless ($target_region_set_name) {
+        $target_region_set_name = $id->target_region_set_name;
+    } else {
+        if ($id->target_region_set_name ne $target_region_set_name) {
+            $multiple_trsn = 1;
+        }
+    }
+}
+
+$inputs->{tumor_bams} = \@tumor_bams;
+$inputs->{tumor_readgroups} = \@tumor_rg_fields;
+
+$inputs->{normal_bams} = \@normal_bams;
+$inputs->{normal_readgroups} = \@normal_rg_fields;
+
+my ($roi_input, @extra_roi) = grep { $_->name eq 'region_of_interest_set_name' } @inputs;
+if (@extra_roi) {
+    $build->fatal_message('multiple inputs found for reference');
+}
+my $fl;
+
+if ($roi_input) {
+    $fl = Genome::FeatureList->get(name => $roi_input->value_id)
+        or $build->fatal_message('no ROI found for input');
+} elsif ($multiple_trsn) {
+    $build->fatal_message('Multiple target_region_set_name found on instrument data. Please specify a "region_of_interest_set_name" input explicitly.');
+} elsif ($target_region_set_name) {
+    $fl = Genome::FeatureList->get(name => $target_region_set_name)
```

Comment:
This needs to handle newer LIMS target sets where the ID is tossed in the instrument data's `target_region_set_name` column.
```suggestion
    $fl = Genome::FeatureList->get(name => $target_region_set_name) // Genome::FeatureList->get(id => $target_region_set_name)
```